ENIGMAsystem/SHELL/Widget_Systems/xlib/kdialog.cpp

```diff
@@ -544,4 +536,4 @@ void message_set_caption(string title) {
   if (error_caption.empty()) error_caption = "Error";
 }
 
-} // namespace enigma_user
+} // namespace kdialog
```

Comment:
After you close out the class (mind the semicolon), you'll need to declare the engine as used in the other files:
```cpp
};  // class KDialogWidgets

// Declares storage for our implementation and exposes it as a system.
static KDialogWidgets kdialog_widgets_impl;
namespace enigma {
  CommandLineWidgetEngine *kdialog_widgets = &kdialog_widgets_impl;
}
```